sdk/azidentity/managed_identity_credential.go

```diff
@@ -54,5 +54,6 @@ func (c *ManagedIdentityCredential) GetToken(ctx context.Context, opts azcore.To
 
 // AuthenticationPolicy implements the azcore.Credential interface on ManagedIdentityCredential.
 func (c *ManagedIdentityCredential) AuthenticationPolicy(options azcore.AuthenticationPolicyOptions) azcore.Policy {
+	options.Options.Scopes = []string{scopesToResource(options.Options.Scopes[0])}
```

Comment:
Why does the `/.default` need to be removed from the scope?  If this is the only place it happens I suggest doing that here inline and remove the function.  Also add a comment why this is necessary.

Reply:
It does, this credential expects a resource string and not a scope. So I'll just put the code directly in the authpolicy func

Reply:
Is this specific to IMDS, i.e. does it use a slice of resources and not a slice of scopes?  Wondering if we should have a separate field for slice of resources.  At the very least, if this is unique behavior for IMDS and we want to use the `Scopes` slice as a resources slice we should comment that.

Reply:
It will expect a slice of resource strings that must match the URI exactly as it is specified in Azure AD. 
An example request from the docs: $response = Invoke-WebRequest -Uri 'http://169.254.169.254/metadata/identity/oauth2/token?api-version=2018-02-01&resource=https://management.azure.com/' -Method GET -Headers @{Metadata="true"}

And the note that went with it: 
The value of the "resource" parameter must be an exact match for what is expected by Azure AD. When using the Azure Resource Manager resource ID, you must include the trailing slash on the URI.

I could document in the credential's info that this credential expects a slice of resources instead of a slice of scopes

Also, if we were to add a specific field for resources it would be on TokenRequestOptions which is in AuthenticationPolicyOptions

Reply:
Here are examples of how to send MSI requests in each environment: 
IMDS: curl 'http://169.254.169.254/metadata/identity/oauth2/token?resource=https://vault.azure.net&api-version=2018-02-01' -H "Metadata: true"

AppService: curl "$MSI_ENDPOINT?resource=https://vault.azure.net&api-version=2017-09-01" -H "secret: $MSI_SECRET"

CloudShell: curl -H 'Metadata: true' -d 'resource=https://vault.azure.net' $MSI_ENDPOINT